crates/oxc_transformer/src/typescript/mod.rs

```diff
@@ -23,4 +23,73 @@ impl<'a> TypeScript<'a> {
             params.items.remove(0);
         }
     }
+
+    /// * Remove the top level import / export statements that are types
+    /// * Adds `export {}` if all import / export statements are removed, this is used to tell
+    /// downstream tools that this file is in ESM.
+    pub fn transform_program(&self, program: &mut Program<'a>) {
+        let mut needs_explicit_esm = false;
+
+        for stmt in program.body.iter_mut() {
+            if let Statement::ModuleDeclaration(module_decl) = stmt {
+                needs_explicit_esm = true;
+                match &mut **module_decl {
+                    ModuleDeclaration::ExportNamedDeclaration(decl) => {
+                        decl.specifiers.retain(|specifier| specifier.export_kind.is_value());
+                    }
+                    ModuleDeclaration::ImportDeclaration(decl) => {
+                        if let Some(specifiers) = &mut decl.specifiers {
+                            specifiers.retain(|specifier| match specifier {
+                                ImportDeclarationSpecifier::ImportSpecifier(s) => {
+                                    s.import_kind.is_value()
+                                }
+                                _ => false,
+                            });
+                        }
+                    }
+                    _ => {}
+                }
+            }
+        }
+
+        program.body.retain(|stmt| match stmt {
+            Statement::ModuleDeclaration(module_decl) => match &**module_decl {
+                ModuleDeclaration::ImportDeclaration(decl) => {
+                    if decl.import_kind.is_type() {
+                        return false;
+                    }
+                    if decl.specifiers.as_ref().is_some_and(|specifiers| specifiers.is_empty()) {
+                        return false;
+                    }
+                    true
+                }
+                ModuleDeclaration::ExportNamedDeclaration(decl) => {
+                    if decl.export_kind.is_type() {
+                        return false;
+                    }
+                    if decl.declaration.is_none() && decl.specifiers.is_empty() {
+                        return false;
+                    }
+                    true
+                }
+                _ => true,
+            },
+            _ => true,
+        });
+
+        if needs_explicit_esm
+            && (program.body.is_empty()
+                || !program.body.iter().any(|s| matches!(s, Statement::ModuleDeclaration(_))))
```

Comment:
in math, 
```
∀ s, s ∈ Statement | s ∉ ModuleDeclaration
```
is true for empty body

Reply:
I always get confused by which one to use, thank you for the correction, you may change the code.